Civi/Token/TokenCompatSubscriber.php

```diff
@@ -122,7 +122,7 @@ public function onRender(TokenRenderEvent $e) {
     $e->string = \CRM_Utils_Token::replaceDomainTokens($e->string, $domain, $isHtml, $e->message['tokens'], $useSmarty);
 
     if (!empty($e->context['contact'])) {
-      \CRM_Utils_Token::replaceGreetingTokens($e->string, $e->context['contact'], $e->context['contact']['contact_id'], NULL, $useSmarty);
```

Comment:
I don't want to bring it into scope for getting this merged but I've been digging into these 2 lines. It seems they basically do the same thing with subtle differences - since both actually 'replaceContactTokens' in the string. 

Despite the name the way we use `replaceGreetingTokens` is not to replace greeting tokens - it is to replace contact tokens (generally).

The differences seem to be:

- the extent to which they replace 'blank' tokens - the first line DOES the second doesn't, 
- the extent to which they attempt to optimise by early exiting if tokens are resolved and
- the extent to which they attempt to load the missing values in the contact

Test cover is good (#21387) & this function & `replaceGreetingTokens` are the only 2 places still calling `replaceContactTokens` so I think the goal would be to consolidate the replacment on this class & call this class from `replaceGreetingTokens` & deprecate `replaceContactTokens`.